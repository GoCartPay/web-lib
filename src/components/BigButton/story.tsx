import React from 'react'
import { MuiButton } from './index'
import { ButtonProps } from '@mui/material/Button'
import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt';
import Box from '@mui/material/Box';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/BigButton',
  component: MuiButton,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' }
  },
  parameters: {
    design: {
      type: 'figma',
      url:
        'https://www.figma.com/file/GDkKRn0hdzPqwrILgdvKr1/Enrollment-with-SPRUCE?node-id=3900%3A32191',
    },
  },
}

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args: ButtonProps & { labelText: string }) => {
  return (
    <Box>
      <div>
        <MuiButton size="large" labelText="Label" {...args} />
      </div>
      <div>
        <MuiButton variant="outlined" size="large" labelText="Label" {...args} />
      </div>
      <div>
        <MuiButton variant="contained" size="large" labelText="Label" {...args} />
      </div>
      <div>
        <MuiButton variant="outlined" endIcon={<ArrowRightAltIcon />} labelText="Label" {...args} />
        <MuiButton variant="contained" endIcon={<ArrowRightAltIcon />} labelText="Label" {...args} />
      </div>
    </Box>
  )
}

export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  labelText: 'Button'
}
