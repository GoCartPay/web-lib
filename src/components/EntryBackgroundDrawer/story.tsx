import React, { useEffect } from 'react'

import { EntryBackgroundDrawer } from './index'
import { SwipeableDrawerProps } from '@mui/material/SwipeableDrawer'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/Components/EntryBackgroundDrawer',
  component: EntryBackgroundDrawer,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
  }
}

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args: SwipeableDrawerProps & { content: any, open: boolean, hasLoader: boolean }) => {

  const [isOpen, setIsOpen] = React.useState(() => args.open);

  useEffect(() => {
    if (args.open !== isOpen) {
      setIsOpen(args.open);
    }
  }, [args.open]);

  const onClose = () => {
    setIsOpen(false);
  }

  return (
  <>
    <EntryBackgroundDrawer onClose={() => onClose()} open={isOpen} content={args.content} hasLoader={true} />
  </>
  )
}


export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  content: '',
  hasLoader: true,
  open: true
}
