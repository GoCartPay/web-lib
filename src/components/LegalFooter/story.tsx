import React from 'react'

import LegalFooter from './index'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/Components/LegalFooter',
  component: LegalFooter,
}

type LegalFooterStoryProps = {
  isScreenSizeMobile: boolean
}
// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args: LegalFooterStoryProps) => {
  const { isScreenSizeMobile } = args;

  return (
    <div style={{ maxWidth: isScreenSizeMobile ? '288px' : '327px', margin: '0px auto' }}>
      <LegalFooter />
    </div>
  )
}

export const Default = Template.bind({})

export const MobileScreenSize = Template.bind({});

MobileScreenSize.args = {
  isScreenSizeMobile: true
}
