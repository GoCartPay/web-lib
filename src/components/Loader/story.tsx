import React, { useEffect } from 'react'

import { Loader } from './index'
import { makeStyles } from '@mui/material/styles'
import Box from '@mui/material/Box';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/Components/Loader',
  component: Loader,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    
  },
  parameters: {
    design: {
      type: 'figma',
      url:
        'https://www.figma.com/file/GDkKRn0hdzPqwrILgdvKr1/Enrollment-with-SPRUCE?node-id=3931%3A32528',
    },
  },
}

const useStyles: any = makeStyles({
  container: {
    backgroundColor: '#fff',
    position: 'absolute',
    bottom: '0',
    width: '100%',
    maxWidth: '430px',
    padding: '48px',
    borderRadius: '20px 20px 0 0',
    top: '180px',
  },
});

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args: any & { loading: boolean }) => {

  const [loading, setLoading] = React.useState(args.loading);

  useEffect(() => {
    if (args.loading !== loading) {
      setLoading(args.loading);
    }
  }, [args.loading]);

  return (
  <Box sx={{ backgroundColor: 'black', padding: '30px' }}>
    {loading && <Loader onComplete={() => { setLoading(false) }} /> }
  </Box>
  )
}

export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  loading: true
}
