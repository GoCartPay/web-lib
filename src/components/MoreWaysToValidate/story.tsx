import React, { useState } from 'react';
import MoreWaysToValidate from '.';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/Components/MoreWaysToValidate',
  component: MoreWaysToValidate,
}

const userPhone = '(•••) ••••• 554';
const userEmail = 'p••l@philj.com';

const radioOptions = [
    { value: 'SMS', label: `Send to ${userPhone}` },
    { value: 'email', label: `Send to ${userEmail}`}
];

const Template = () => {
  const [isOpen, setIsOpen] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [resendSuccess, setResendSuccess] = useState(false);
  const [authType, setAuthType] = useState('SMS');

  const handleResendOTP = (val: string) => {
    setIsLoading(true);
    setAuthType(val);
    setTimeout(() => {
      setIsLoading(false);
      setResendSuccess(true);
    },1000);
  };

  const handleEnterNewCode = () => {
    setIsOpen(false);
    setResendSuccess(false);
  };

  if (!isOpen) return <button onClick={() => setIsOpen(true)}>Open</button>;

  return (
    <div style={{ maxWidth: '327px', margin: '0px auto' }}>
      <MoreWaysToValidate
        isLoading={isLoading}
        isOpen={isOpen}
        radioOptions={radioOptions}
        onCancel={() => setIsOpen(false)}
        onEnterNewCode={handleEnterNewCode}
        onResend={handleResendOTP}
        showSuccess={resendSuccess}
        successMessage={`Sent to ${authType === 'SMS' ? userPhone : userEmail}`}
      />
    </div>
  )
};

export const Default = Template.bind({})