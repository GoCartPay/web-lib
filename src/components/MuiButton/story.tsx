import React from 'react'

import MuiButton from './index'
import { ButtonProps } from '@mui/material/Button'
import DeleteIcon from '@mui/icons-material/Delete';
import SendIcon from '@mui/icons-material/Send';
import Stack from '@mui/material/Stack';
import Box from '@mui/material/Box';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/MuiButton',
  component: MuiButton,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' }
  }
}

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args: ButtonProps & { labelText: string }) => (
  <Box sx={{ '& button': { m: 1 } }}>
    <div>
      <MuiButton size="small">Small</MuiButton>
      <MuiButton size="medium">Medium</MuiButton>
      <MuiButton size="large">Large</MuiButton>
    </div>
    <div>
      <MuiButton variant="outlined" size="small">
        Small
      </MuiButton>
      <MuiButton variant="outlined" size="medium">
        Medium
      </MuiButton>
      <MuiButton variant="outlined" size="large">
        Large
      </MuiButton>
    </div>
    <div>
      <MuiButton variant="contained" size="small">
        Small
      </MuiButton>
      <MuiButton variant="contained" size="medium">
        Medium
      </MuiButton>
      <MuiButton variant="contained" size="large">
        Large
      </MuiButton>
    </div>
    <div>
      <Stack direction="row" spacing={2}>
        <MuiButton variant="outlined" startIcon={<DeleteIcon />}>
          Delete
        </MuiButton>
        <MuiButton variant="contained" endIcon={<SendIcon />}>
          Send
        </MuiButton>
      </Stack>
    </div>
  </Box>

)

export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  labelText: 'Button'
}
