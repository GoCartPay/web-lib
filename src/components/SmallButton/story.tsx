import React from 'react'
import { ButtonProps } from '@mui/material/Button'

import SmallButton from './index'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/Components/SmallButton',
  component: SmallButton,
}

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args: ButtonProps & { labelText: string }) => (
  <SmallButton {...args} />
)

export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  labelText: 'Label'
}

export const WithBorder = Template.bind({});

WithBorder.args = {
  ...Default.args,
  variant: 'outlined',
  labelText: 'Label'
}