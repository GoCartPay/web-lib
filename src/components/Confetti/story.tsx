import React from 'react'
import { MuiButton } from '../MuiButton/index'
import ConfettiExplosion from 'react-confetti-explosion';
import Box from '@mui/material/Box';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Spruce/Components/Confetti',
  component: ConfettiExplosion,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' }
  },
  parameters: {
    design: {
      type: 'figma',
      url:
        'https://www.figma.com/file/GDkKRn0hdzPqwrILgdvKr1/Enrollment-with-SPRUCE?node-id=3931%3A32528',
    },
  },
}

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args: any) => {
  const [isExploding, setIsExploding] = React.useState(false);

  return (
    <Box>
      <MuiButton size="large" labelText="Splode" onClick={() => { setIsExploding(true) }} />
      <MuiButton size="large" labelText="Reset" onClick={() => { setIsExploding(false) }} />
      <Box sx={{
        'div.container': {
          marginLeft: '150px',
          height: '400px',
          width: '400px',
          backgroundColor: '#333'
        }
      }}>
        <div className="container">
          {isExploding && <ConfettiExplosion {...args} />}
        </div>
        </Box>
    </Box>
  )
}

export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  particleCount: 150,
  particleSize: 8,
  duration: 2000,
  colors: [
    '#22974A',
    '#2AD062',
    '#5EF391',
    '#CBFFDD'
  ],
  force: 0.75,
  floorHeight: 300,
  floorWidth: 425
}
