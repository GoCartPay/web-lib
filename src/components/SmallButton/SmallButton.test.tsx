import React from 'react';
import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import * as SmallButton from  './story';

// this seems like this is more of testing the story rather than testing the component itself
const { Default, WithBorder } = composeStories(SmallButton);

const mockOnClick = jest.fn();

describe('Small button behaves correctly',  () => {

  it('renders without crashing', () => {
    render(
      // @ts-expect-error this does have a call signature 
      <Default/>
    );
    expect(screen.getByRole('button')).toBeTruthy();
  });
  it('matches snapshot for a text button', () => {
    // @ts-expect-error this does have a call signature 
    const { container } = render(<Default/>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('allows for an event to fire onClick', () => {
    render(
      // @ts-expect-error this does have a call signature 
      <Default 
        onClick={mockOnClick}
      />
    );
    const smallBtnElem = screen.getByRole('button');
    userEvent.click(smallBtnElem);
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })

  it('Outlined variant renders button with border state and matches snapshot', () => {
    const { container } = render(
      // @ts-expect-error this does have a call signature 
      <WithBorder />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
})