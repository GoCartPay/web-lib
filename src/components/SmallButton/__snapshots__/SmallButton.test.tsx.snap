// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small button behaves correctly Outlined variant renders button with border state and matches snapshot 1`] = `
<button
  class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-fhf16q-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Label
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Small button behaves correctly matches snapshot for a text button 1`] = `
<button
  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1y2hhty-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Label
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
